benchmarks/benchmark_serving.py

```diff
@@ -163,21 +166,23 @@ async def benchmark(
     request_rate: float,
 ) -> None:
     tasks: List[asyncio.Task] = []
+    pbar = tqdm(total=len(input_requests))
     async for request in get_request(input_requests, request_rate):
         prompt, prompt_len, output_len = request
         task = asyncio.create_task(
             send_request(backend, model, api_url, prompt, prompt_len,
-                         output_len, best_of, use_beam_search))
+                         output_len, best_of, use_beam_search, pbar))
         tasks.append(task)
-    await tqdm.gather(*tasks)
+    await asyncio.gather(*tasks)
+    pbar.close()
 
 
 def main(args: argparse.Namespace):
     print(args)
     random.seed(args.seed)
     np.random.seed(args.seed)
 
-    api_url = f"http://{args.host}:{args.port}{args.endpoint}"
+    api_url = f"{args.protocol}://{args.host}:{args.port}{args.endpoint}"
```

Comment:
What about just removing all this custom assembling and allowing a `url`? Because this won't work if you don't have a port?

Reply:
You could consolidate all the arguments, not sure what design upstream would prefer.

Also, there is always a port. i.e. if you use HTTP the default port is 80 and for HTTPS the default port is 443